code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ProductRatingNotificationBuilder.kt

```diff
@@ -0,0 +1,226 @@
+        trackerActivityClass: Class<out Activity>?,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?
+    ): NotificationCompat.Builder {
+        Log.trace(
+            LOG_TAG,
+            SELF_TAG,
+            "Building a rating template push notification."
+        )
+        val packageName = context.packageName
+        val smallLayout = RemoteViews(packageName, R.layout.push_template_collapsed)
+        val expandedLayout = RemoteViews(packageName, R.layout.push_template_product_rating_expanded)
+
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val channelIdToUse: String = notificationManager.createNotificationChannelIfRequired(
+            context,
+            pushTemplate
+        )
+
+        // create the notification builder with the common settings applied
+        val notificationBuilder = AEPPushNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            channelIdToUse,
+            trackerActivityClass,
+            smallLayout,
+            expandedLayout,
+            R.id.rating_expanded_layout
+        )
+
+        // set the image on the notification
+        val imageUri = pushTemplate.imageUrl
+        val downloadedImageCount = PushTemplateImageUtils.cacheImages(listOf(imageUri))
+
+        if (downloadedImageCount == 0) {
+            Log.trace(
+                LOG_TAG,
+                SELF_TAG,
+                "No image found for rating push template."
+            )
+            expandedLayout.setViewVisibility(R.id.expanded_template_image, View.GONE)
+        } else {
+            expandedLayout.setImageViewBitmap(
+                R.id.expanded_template_image,
+                PushTemplateImageUtils.getCachedImage(imageUri)
+            )
+        }
+
+        // populate the rating icons
+        populateRatingIcons(
+            context,
+            broadcastReceiverClass,
+            expandedLayout,
+            pushTemplate,
+            packageName,
+            channelIdToUse
+        )
+
+        // check if confirm button needs to be shown
+        if (pushTemplate.ratingSelected >= 0) {
+            expandedLayout.setViewVisibility(R.id.rating_confirm, View.VISIBLE)
+            expandedLayout.setNotificationTitleTextColor(
+                pushTemplate.titleTextColor,
+                R.id.rating_confirm
+            )
+
+            // add pending intent for confirm click
+            // sticky is set to false as the notification will be dismissed after confirm click
+            val selectedRatingAction = pushTemplate.ratingActionList[pushTemplate.ratingSelected]
+            expandedLayout.setRemoteViewClickAction(
+                context,
+                trackerActivityClass,
+                R.id.rating_confirm,
+                selectedRatingAction.link,
+                selectedRatingAction.type,
+                pushTemplate.tag,
+                false
```

Comment:
sticky is a configurable setting for rating push templates. i think we should read the KVP from the payload and set this accordingly.

Reply:
Having `sticky = true` once confirm is clicked would allow the end user to select rating multiple times for the same product. I confirmed with Don the other day that we will be dismissing the notification once confirm is clicked.